barretenberg/cpp/src/barretenberg/stdlib/primitives/bigfield/bigfield_impl.hpp

```diff
@@ -1313,7 +1356,9 @@ bigfield<Builder, T> bigfield<Builder, T>::mult_madd(const std::vector<bigfield>
             const auto [quotient_1024, remainder_1024] =
                 (sum_of_constant_products + add_right_constant_sum).divmod(modulus);
             ASSERT(!fix_remainder_to_zero || remainder_1024 == 0);
-            return bigfield(ctx, uint256_t(remainder_1024.lo.lo));
+            auto result = bigfield(ctx, uint256_t(remainder_1024.lo.lo));
```

Comment:
it would be interesting to see if we can include the setting of the origin tag in the constructor because a bug that can happen in the current design is we create a new method and forget altogether about origin tag. Or you forget to set it in some function during your work

Reply:
There's a lot we can do later, once the tags are in every primitive, for example do instant_death for every from_witness to force the circuit writer to account for elements created this way.